branchLayoutApi/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -51,4 +51,58 @@ public IBranchLayout slideOut(String branchName) throws BranchLayoutException {
       return List.of(entry.withChildren(children.flatMap(child -> slideOut(child, entryToSlideOut))));
     }
   }
+
+  @Override
+  public IBranchLayout slideIn(String parentBranchName, IBranchLayoutEntry entryToSlideIn) throws BranchLayoutException {
+    var parentEntry = findEntryByName(parentBranchName).getOrNull();
+    if (parentEntry == null) {
+      throw new BranchLayoutException("Parent branch entry '${parentBranchName}' does not exists");
+    }
+    var entry = findEntryByName(entryToSlideIn.getName());
+    var entryAlreadyExists = entry.isDefined();
+
+    if (entry.map(e -> isDescendant(e, parentEntry)).getOrElse(false)) {
+      throw new BranchLayoutException(
+          "Entry '${parentEntry.getName()}' is an descendant of entry '${entryToSlideIn.getName()}'");
```

Comment:
`a descendant`

Reply:
While you're here... there's also `a unambiguous` in gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:311 ;D